### PR TITLE
 feat: Wrap-around selection for TUI.

### PR DESCRIPTION
**Pull request by anthonyshew**
### Description

We previously didn't let you "wrap" your selection to the selected task if you were at the top or bottom of the task list. Now we do!

### Testing Instructions

Run some tasks with the TUI and:
-  Press ↑ when you have the top item in the list selected.
- Press ↓ when you have the bottom item in the list selected.

### CODE REVIEW

File: crates/turborepo-ui/src/tui/app.rs [hunk 1]
```diff
@@ -886,6 +886,8 @@ mod test {
         app.next();
         assert_eq!(app.scroll.selected(), Some(2), "scroll moves forwards");
         app.next();
+        assert_eq!(app.scroll.selected(), Some(0), "scroll stays in bounds");
```
Comment (chris-olszewski) [inline crates/turborepo-ui/src/tui/app.rs:889]:
```suggestion
        assert_eq!(app.scroll.selected(), Some(0), "scroll wraps");
```

File: crates/turborepo-ui/src/tui/app.rs [hunk 2]
```diff
@@ -142,21 +142,21 @@ impl<W> App<W> {
...
-        let i = match self.selected_task_index {
-            0 => 0,
-            i => i - 1,
-        };
-        self.selected_task_index = i;
-        self.scroll.select(Some(i));
-        self.has_user_scrolled = true;
+        let num_rows = self.tasks_by_status.count_all();
+        if num_rows > 0 {
+            self.selected_task_index = (self.selected_task_index + num_rows - 1) % num_rows;
```
Comment (chris-olszewski) [inline crates/turborepo-ui/src/tui/app.rs:156]: This one is totally up to you if you think explicitly handling the underflow case is easier to reason about than leveraging modulos.
```suggestion
            self.selected_task_index = self.selected_task_index.checked_sub(1).unwrap_or(num_rows - 1);
```

**Review by chris-olszewski — APPROVED** [review]
LFG

File: crates/turborepo-ui/src/tui/app.rs (hunk 2 above)
Comment (anthonyshew) [reply crates/turborepo-ui/src/tui/app.rs:156]: Classic me not knowing methods I have available.

Comment (chris-olszewski) [reply crates/turborepo-ui/src/tui/app.rs]: I'm an `Option` addict, I love how clean the control flow that methods provide is.

**Pull request merged**